src/firefly/js/charts/dataTypes/FireflyGenericData.js

```diff
@@ -105,7 +105,7 @@ export function createChartTblRequest(chartId, traceNum, tablesource) {
     let sreq = cloneRequest(request, {
         startIdx: 0,
         pageSize: MAX_ROW,
-        inclCols: Object.entries(mappings).filter(([k, v]) => !numberOrArrayProps.includes(k) || Number.isNaN(parseFloat(v))).map(([k, v]) => {
+        inclCols: Object.entries(mappings).filter(([k, v]) => !(numberOrArrayProps.includes(k) && !isNaN(v))).map(([k, v]) => {   // if field can be a number and is a number, then skip
```

Comment:
A couple of thoughts just to make sure this is the change you want to make:

- `Number.isNan` I think requires a number while `isNaN` converts to a number.
- `parseFloat` will convert the number part of a string and throw out the rest so `’123.45.abc’` will be `123.45`
- so the change is a little more than simplifying the code to `&& !`

I don’t know if this really matters but it could affect the processing of the expression a little.

Reply:
Yes.  `parseFloat` was the cause of the problem.  '2 * ra' returns 2, which is a number, therefore is excluded from the filter.  I use `isNaN` because it's more strict when parsing.  I also changed the condition to make it more precise.